gui.py

```diff
@@ -0,0 +1,284 @@
+
+
+        # create vbox layouts and add to setlayout groupbox
+        self.vboxPlaylist = QtWidgets.QVBoxLayout()
+        self.vboxPlaylist.setContentsMargins(0, 0, 0, 0)
+        self.vboxPlaylist.setSpacing(2)
+        self.groupBoxPlaylist.setLayout(self.vboxPlaylist)
+
+        self.vboxSongs = QtWidgets.QVBoxLayout()
+        self.vboxSongs.setContentsMargins(0, 0, 0, 0)
+        self.groupBoxSongs.setLayout(self.vboxSongs)
+
+        self.vboxReleases = QtWidgets.QVBoxLayout()
+        self.vboxReleases.setContentsMargins(0, 0, 0, 0)
+        self.groupBoxReleases.setLayout(self.vboxReleases)
+
+        # create treeview
+        self.treeViewPlaylist = QtWidgets.QTreeView()
+        self.treeViewPlaylistModel = QtGui.QStandardItemModel()
+        self.treeViewPlaylist.setModel(self.treeViewPlaylistModel)
+        self.treeViewPlaylist.clicked.connect(self.treeview_on_clicked)
+        self.vboxPlaylist.addWidget(self.treeViewPlaylist)
+        self.rootNode = self.treeViewPlaylistModel.invisibleRootItem()
+
+        # create tableviewplaylistsongs
+        self.TableViewPlaylistSongs = GuiTableView(self)
+        self.vboxSongs.addWidget(self.TableViewPlaylistSongs)
+
+        # create tableviewreleases
+        self.TableViewReleases = GuiTableView(self)
+        self.TableViewReleases.clicked.connect(self.tableviewreleases_on_click)
+        self.vboxReleases.addWidget(self.TableViewReleases)
+
+        # create vbox formlayout for buttons and edit box
+        self.vboxFormLayout = QtWidgets.QFormLayout()
+
+        # lineedit playlistname
+        self.lineEditAddPlaylistName = QtWidgets.QLineEdit()
+        self.lineEditAddPlaylistName.setPlaceholderText('Playlist name')
+        self.vboxFormLayout.addRow(self.lineEditAddPlaylistName)
+
+        # lineedit playlistvenue
+        self.lineEditAddPlaylistVenue = QtWidgets.QLineEdit()
+        self.lineEditAddPlaylistVenue.setPlaceholderText('Venue')
+        self.vboxFormLayout.addRow(self.lineEditAddPlaylistVenue)
+
+        # lineedit playlistdate
+        self.lineEditAddPlaylistDate = QtWidgets.QLineEdit()
+        self.lineEditAddPlaylistDate.setPlaceholderText('2021-01-01')
+        self.vboxFormLayout.addRow(self.lineEditAddPlaylistDate)
+
+        # pushbuttonadd
+        self.pushButtonAddPlaylist = QtWidgets.QPushButton()
+        self.pushButtonAddPlaylist.setText('Add')
+        self.pushButtonAddPlaylist.clicked.connect(self.treeview_pushbutton_add_playlist)
+
+        #pushbuttondel
+        self.pushButtonDelPlaylist = QtWidgets.QPushButton()
+        self.pushButtonDelPlaylist.setText('Delete')
+        self.pushButtonDelPlaylist.clicked.connect(self.treeview_pushbutton_del_playlist)
+        self.vboxFormLayout.addRow(self.pushButtonAddPlaylist, self.pushButtonDelPlaylist)
+
+        # add formlayout to playlist boxlayout
+        self.vboxPlaylist.addLayout(self.vboxFormLayout)
+
+        self.conf = Config()
```

Comment:
Not sure if this is possible right now or a good idea but try to move this line up to the start of `MainWindow.__init__`
Reasons and for now actually only a quickfix: Config does a lot of things including creating .discodos directory in the users's home, or when running packaged in other directories the user would expect application configuration files. more on that later. 

Main reason for heaving it on the very top before even the gui is initialized is: If it is missing a discogs token, it prompts for it on the shell. at the end of the day this prompt should be made by the gui to. Not sure how and when I/we do that but for now try to move the config stuff to the very top before the window starts.

Another reason why it's important: you save settings.ini in the discodos/qt folder, which in packaged installations is a system folder where the application is installed but the executing user does not have access, thus this will fail. You can try to save settings.ini in the same path Config() sets for the discobase.db file - you will find the path as self.conf.discodos_data. Maybe you give it a more distinguishable name then too: gui_settings.ini